pysnmp/meta.yaml

```diff
@@ -0,0 +1,90 @@
+{% set name = "pysnmp" %}
+{% set version = "4.4.2" %}
+{% set file_ext = "tar.gz" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "aaed6e807c26c3a38257daeb851b7a4354df011aa00ed77bd48ef3c4dd8b467b" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.{{ file_ext }}
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  {{ hash_type }}: {{ hash_value }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
+    - python
+    - pyasn1 >=0.2.3
+    - pysmi
+    - pycryptodomex
+
+test:
+  imports:
+    - pysnmp
+    - pysnmp.carrier
+    - pysnmp.carrier.asyncio
+    - pysnmp.carrier.asyncio.dgram
+    - pysnmp.carrier.asyncore
+    - pysnmp.carrier.asyncore.dgram
+    - pysnmp.carrier.asynsock
+    - pysnmp.carrier.asynsock.dgram
+    - pysnmp.carrier.twisted
+    - pysnmp.carrier.twisted.dgram
+    - pysnmp.entity
+    - pysnmp.entity.rfc3413
+    - pysnmp.entity.rfc3413.oneliner
+    - pysnmp.hlapi
+    - pysnmp.hlapi.asyncio
+    - pysnmp.hlapi.asyncore
+    - pysnmp.hlapi.asyncore.sync
+    - pysnmp.hlapi.asyncore.sync.compat
+    - pysnmp.hlapi.twisted
+    - pysnmp.proto
+    - pysnmp.proto.acmod
+    - pysnmp.proto.api
+    - pysnmp.proto.mpmod
+    - pysnmp.proto.proxy
+    - pysnmp.proto.secmod
+    - pysnmp.proto.secmod.eso
+    - pysnmp.proto.secmod.eso.priv
+    - pysnmp.proto.secmod.rfc3414
+    - pysnmp.proto.secmod.rfc3414.auth
+    - pysnmp.proto.secmod.rfc3414.priv
+    - pysnmp.proto.secmod.rfc3826
+    - pysnmp.proto.secmod.rfc3826.priv
+    - pysnmp.proto.secmod.rfc7860
+    - pysnmp.proto.secmod.rfc7860.auth
+    - pysnmp.smi
+    - pysnmp.smi.mibs
+    - pysnmp.smi.mibs.instances
+  requires:
+    - twisted
```

Comment:
Is `twisted` considered an optional requirement? I see it imported in:

https://github.com/etingof/pysnmp/blob/master/pysnmp/carrier/twisted/dgram/base.py#L8

and other locations.

Reply:
They have listed it in devel requirements: https://github.com/etingof/pysnmp/blob/master/devel-requirements.txt
twisted is pulls in a lot of dependencies. 
What's CF's policy on optional dependencies?

Reply:
I'm not sure if there is an official policy. I maintain the `apscheduler` feedstock that has optional twisted, asyncio, etc backends. I don't list them as requirements, but that library handles missing optional requirements gracefully:

https://github.com/agronholm/apscheduler/blob/master/apscheduler/schedulers/twisted.py#L8-L11

giving the user an informative message. I'm not familiar with this library, but I think it's fine to leave out `twisted` if it's optional (i.e you can run the package without it, but it's only used in a particular mode that the user is explicitly choosing to use twisted).

Reply:
I am leaving it out from the 'run' section for now. Maybe someday conda will officially support optional dependencies, then we can add it.